sips/sip-011/sip-011_ustx-marketplace-standard.md

```diff
@@ -0,0 +1,427 @@
+        (unlist-in-ustx (uint) (response bool uint))
+        (buy-in-ustx (uint  <commission-trait>) (response bool uint))
+        (get-asset () (response {fq-contract: string, asset-class: string} uint))
+    )
+)
+
+(define-trait commission
+    (
+        (pay (uint uint) (response bool uint))
+    )
+)
+```
+
+## Security
+As `commission-traits` can call any functions in the name of the tx-sender, it is important that a web app only offers commission contracts that are well understood. In particular, appropriate post-conditions have to be created.
+
+If asset contracts want to control trades they have to restrict which operators are approved. Note, that royalties to an artist of an NFT can be part of the commission if agreed with the marketplace. They can also be implemented in the NFT directly or as a seperate trait. See reference implementation list-in-ustx.
+
+## Use of native asset functions
+Werner: not sure what to do here (section just a copy from sip009)
+
+Although it is not possible to mandate in a Clarity trait, contract implementers must define at least one built-in native non-fungible [asset class](https://app.sigle.io/friedger.id/FDwT_3yuMrHDQm-Ai1OVS) that are provided as Clarity primitives. This allows clients to use Post Conditions (explained below), and takes advantages of other benefits, like native support for these asset balances and transfers through `stacks-blockchain-api`. The reference implementations included in this SIP use the native asset primitives, and provide a good boilerplate for their usage.
+
+The native asset functions include:
+
+- `define-non-fungible-token`
+- `nft-burn?`
+- `nft-get-owner?`
+- `nft-mint?`
+- `nft-transfer?`
+
+The following requirements for using native asset functions are defined:
+
+### Transfer
+
+If the `transfer` function is called from a client without a [post-condition](https://docs.blockstack.org/understand-stacks/transactions#post-conditions) in deny mode or without any NFT condition about a changed owner, then the function call must fail with `abort_by_post_condition`.
+
+### Recommendations for displaying commissions
+
+It is advised to show the user what the listing price will be on the platform used and how much of that will go to the interested parties when sold such as the seller, the artist and a marketplace.
+
+It is recommended to marketplaces to do the following if a user lists something at 1 STX, don't send 1000000 to list-in-ustx, but rather 1000000/(1+comm) as the price argument.
+
+Some examples of how the user would be informed as displayed in December 2022 on Gamma.io and Tradeport.xyz:
+
+**Example 1 tradeport.xyz**
+
+![image](https://user-images.githubusercontent.com/33360391/208997197-61111244-b597-46fd-9c08-559550a127aa.png)
+
+**Example 2 gamma.io**
+
+![image](https://user-images.githubusercontent.com/33360391/208997558-6dcbdef7-692f-4b41-b7eb-df429c3cfead.png)
+
+### Recommendations for error codes
```

Comment:
This section needs some expansion. It would be a good idea to loop in relevant people on the PR thread